spyder/widgets/ipythonconsole/client.py

```diff
@@ -163,7 +165,10 @@ def stderr_file(self):
         """Filename to save kernel stderr output."""
         if self.connection_file is not None:
             stderr_file = self.kernel_id + '.stderr'
-            stderr_file = osp.join(TEMPDIR, stderr_file)
+            if not self.testing:
+                stderr_file = osp.join(TEMPDIR, stderr_file)
+            else:
+                stderr_file = osp.join(TEMPDIR, u'測試', u'اختبار', stderr_file)

```

Comment:
Please also find a solution similar to the one I posted above to avoid putting this temp directory as part of our source code.